### ISSUE TITLE
False negative: `unspecified-encoding` (`W1514`) not raised for `Path.read_text`

### ISSUE THREAD

**Issue by GideonBear** [PyCQA/pylint#8419]
### Current problem

```py
from pathlib import path

text = Path('file.txt').read_text()
```
Does not raise `unspecified-encoding` like expected.

### Desired solution

Raise `unspecified-encoding` for `Path.read_text` too.

### Additional context

```
pylint 2.17.0
astroid 2.15.0
Python 3.9.15 (main, Nov 11 2022, 13:33:56) 
[GCC 9.4.0]
```

**Comment by DudeNr33** [comment]
I cannot reproduce this with the current `main`.
There is even a functional test for exactly this case:
https://github.com/PyCQA/pylint/blob/958ff856a25c1b6152c6f0a03d7488a6278adebf/tests/functional/u/unspecified_encoding_py38.py#L66

**Comment by GideonBear** [comment]
Sorry, PEBCAK; I didn't import `Path` correctly; my actual problem is this:
```py
from pathlib import Path

a = Path('a')
text = (a / 'b.txt').read_text()
```
Is this something that will be added in the future? Is there an open issue for these kinds of type inferences?

**Comment by Pierre-Sassoulas** [comment]
I can reproduce the new example.

**Comment by RomanValov** [comment]
I have encountered another case where W1514 won't raise:

```python

def read_from(path: pathlib.Path) -> str:
  with path.open('r') as fp:
    return fp.read()

if __name__ == '__main__':
  print(read_from(pathlib.Path(__file__)))
```

I wonder if pylint ever would be able to catch it, in particular:

1) could it take typing annotation into account?
2) could it make cross-function analysis?
